clion/src/main/kotlin/org/rust/clion/debugger/runconfig/RsCLionDebugRunnerUtils.kt

```diff
@@ -5,27 +5,40 @@
 
 package org.rust.clion.debugger.runconfig
 
+import com.intellij.openapi.application.ApplicationInfo
 import com.intellij.openapi.options.ShowSettingsUtil
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.ui.Messages
-import com.intellij.openapi.util.SystemInfo
+import com.intellij.openapi.util.BuildNumber
 import com.jetbrains.cidr.cpp.toolchains.CPPToolchains
 import com.jetbrains.cidr.cpp.toolchains.CPPToolchainsConfigurable
+import com.jetbrains.cidr.toolchains.OSType
 import org.rust.cargo.runconfig.CargoRunStateBase
+import org.rust.debugger.runconfig.RsDebugRunnerUtils
 import org.rust.debugger.runconfig.RsDebugRunnerUtils.ERROR_MESSAGE_TITLE
-import org.rust.debugger.runconfig.RsDebugRunnerUtils.MSVC_IS_NOT_SUPPORTED_MESSAGE
 import org.rust.openapiext.computeWithCancelableProgress
 
 object RsCLionDebugRunnerUtils {
 
     fun checkToolchainSupported(project: Project, state: CargoRunStateBase): Boolean {
-        if (!SystemInfo.isWindows) return true
-        val isMsvc = project.computeWithCancelableProgress("Checking if toolchain is supported...") {
-            "msvc" in state.rustVersion().rustc?.host.orEmpty()
-        }
-        if (isMsvc) {
-            Messages.showErrorDialog(project, MSVC_IS_NOT_SUPPORTED_MESSAGE, ERROR_MESSAGE_TITLE)
-            return false
+        val toolSet = CPPToolchains.getInstance().defaultToolchain?.toolSet ?: return false
+        if (CPPToolchains.getInstance().osType == OSType.WIN) {
+            val host = project.computeWithCancelableProgress("Checking if toolchain is supported...") {
+                state.rustVersion().rustc?.host.orEmpty()
+            }
+            val isMSVCRustToolchain = "msvc" in host
+            val isGNURustToolchain = "gnu" in host
+
+            if (ApplicationInfo.getInstance().build < BuildNumber.fromString("202")!! && isMSVCRustToolchain) {
+                Messages.showErrorDialog(project, RsDebugRunnerUtils.MSVC_IS_NOT_SUPPORTED_MESSAGE, ERROR_MESSAGE_TITLE)
+                return false
+            }
+
+            return when {
+                isGNURustToolchain && toolSet.isMSVC -> false
+                isMSVCRustToolchain && !toolSet.isMSVC -> false
```

Comment:
* when build tool window is enabled. IDE doesn't do any user-visible actions - as a result, if my toolchain is not configured properly, I can't know about it
* when build tool window is disabled (currently, default case). IDE shows me wrong dialog

<img width="429" alt="Screen Shot 2020-06-15 at 16 18 47" src="https://user-images.githubusercontent.com/2539310/84662192-24d76200-af24-11ea-9d01-99c761f55312.png">